keepsorted/options_parser.go

```diff
@@ -0,0 +1,97 @@
+package keepsorted
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+var (
+	boolValues = map[string]bool{
+		"yes":   true,
+		"true":  true,
+		"no":    false,
+		"false": false,
+	}
+	boolString = map[bool]string{
+		true:  "yes",
+		false: "no",
+	}
+	keyRegex = regexp.MustCompile("(^| )(?P<key>[a-z_]+)=")
+)
+
+type parser struct {
+	line string
+}
+
+func (p *parser) popKey() (string, bool) {
+	m := keyRegex.FindStringSubmatchIndex(p.line)
+	if m == nil {
+		return "", false
+	}
+	key := string(keyRegex.ExpandString(nil, "${key}", p.line, m))
+	p.line = p.line[m[1]:]
+	return key, true
+}
+
+func (p *parser) popValue(typ reflect.Type) (reflect.Value, error) {
+	switch typ {
+	case reflect.TypeFor[bool]():
+		val, err := p.popBool()
+		return reflect.ValueOf(val), err
+	case reflect.TypeFor[int]():
+		val, err := p.popInt()
+		return reflect.ValueOf(val), err
+	case reflect.TypeFor[[]string]():
+		val, err := p.popList()
+		return reflect.ValueOf(val), err
+	case reflect.TypeFor[map[string]bool]():
+		val, err := p.popSet()
+		return reflect.ValueOf(val), err
+	}
+
+	panic(fmt.Errorf("unhandled case in switch: %v", typ))
```

Comment:
Why panic when this function can return an error?

Reply:
It feels like programmer error if we ever forget to support a case in this switch. I would rather that fail pretty loudly so we're more likely to catch it at test time instead of an error which might get swalled with a `wantErr: true`-type test